packages/http/libraries/core/src/http/adapter/InversifyHttpAdapter.ts

```diff
@@ -188,19 +195,20 @@ export abstract class InversifyHttpAdapter<
     request: TRequest,
     response: TResponse,
     value: ControllerResponse,
+    statusCode?: HttpStatusCode,
   ): unknown {
     let body: object | string | number | boolean | undefined = undefined;
-    let statusCode: HttpStatusCode | undefined = undefined;
+    let httpStatusCode: HttpStatusCode | undefined = statusCode;
 
     if (value instanceof HttpResponse) {
```

Comment:
My bad for not seeing this before, never use `instanceof` on non native types, a user might end up with multiple versions of the core. Instead, consider an `is` static method the same way `AppError` does

Reply:
Fixed